libsnark/zk_proof_systems/plonk/tests/test_plonk.cpp

```diff
@@ -1087,6 +1087,56 @@ template<typename ppT> void test_plonk_gates_matrix_transpose()
     ASSERT_EQ(gates_matrix_transpose, example.gates_matrix_transpose);
 }
 
+template<typename ppT> void test_plonk_constants_k1_k2()
+{
+    using Field = libff::Fr<ppT>;
+    Field k1, k2;
+    // n = 2^s
+    const size_t n = std::pow(2, k1.s);
+    bool b_valid = false;
+    // load k1,k2 from example circuit
+    plonk_example example;
+    k1 = example.k1;
+    k2 = example.k2;
+    b_valid = plonk_are_valid_constants_k1_k2(n, k1, k2);
```

Comment:
There doesn't seem to be any need to keep `b_valid` around.
```suggestion
    ASSERT_TRUE(plonk_are_valid_constants_k1_k2(n, k1, k2));
```

Reply:
Can you add a comment to explain exactly what's being tested here?  It's not clear to me why we'd go through every $\omega^i$ for the specific case of $i=0$ up to `example.num_gates`.  I'm sure there's a good reason behind this, but it's not necessarily obvious to the reader so a comment here could be helpful.

Reply:
Here I am checking invalid values for `k1,k2` by setting one of them to be equal to an element of `H`, `k1H` or `k2H`. I added an explanation in the comments. Let me know if you think this check is redundant.